packages/core/src/types/module-builder.ts

```diff
@@ -128,44 +128,149 @@ export interface SendDataOptions {
  * @beta
  */
 export interface IgnitionModuleBuilder {
+  /**
+   * Returns an account runtime value that can be used to specify the
+   * account to use for a transaction.
+   *
+   * @param accountIndex - The index of the account to return
+   *
+   * @example
+   * ```
+   * const owner = m.getAccount(1);
+   * const myContract = m.contract("MyContract", { from: owner });
+   * // can also be used anywhere an address is expected
+   * m.send("sendToOwner", owner, { value: 100 });
+   * ```
+   */
   getAccount(accountIndex: number): AccountRuntimeValue;
 
+  /**
+   * Allows you to specify a parameter that can be dynamically set during deployment.
+   *
+   * @param parameterName - The name of the parameter
+   * @param defaultValue - The default value to use if the parameter is not provided
+   *
+   * @example
+   * ```
+   * const amount = m.getParameter("amount", 100);
+   * const myContract = m.contract("MyContract", { value: amount });
+   * ```
+   */
   getParameter<ParamTypeT extends ModuleParameterType = any>(
     parameterName: string,
     defaultValue?: ParamTypeT
   ): ModuleParameterRuntimeValue<ParamTypeT>;
 
+  /**
+   * Deploys a contract.
+   *
+   * @param contractName - The name of the contract to deploy
+   * @param args - The arguments to pass to the contract constructor
+   * @param options - The options for the deployment
```

Comment:
Can we break out the options and document them.

Reply:
I believe they are documented already, but I will double check